hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/AdapterInstanceDrawer.tsx

```diff
@@ -67,62 +56,43 @@ const AdapterInstanceDrawer: FC<AdapterInstanceDrawerProps> = ({
   const { data } = useGetAdapterTypes()
   const { data: allAdapters } = useListProtocolAdapters()
   const { adapterId } = useParams()
-  const [batchData, setBatchData] = useState<JSONPatchDocument | undefined>(undefined)
 
   const { schema, uiSchema, name, logo, isDiscoverable } = useMemo(() => {
     const adapter: ProtocolAdapter | undefined = data?.items?.find((e) => e.id === adapterType)
     const { configSchema, uiSchema, capabilities } = adapter || {}
 
-    // TODO[NVL] This is still a hack; backend needs to provide identification of mapping-related properties
-    const paths = getTopicPaths(configSchema || {})
-    const subIndex = getMainRootFromPath(paths)
-    const hiddenMappingsKey = import.meta.env.VITE_FLAG_ADAPTER_SPLIT_SCHEMA === 'true' ? subIndex : undefined
+    // If the config schema is split across entities, replace the fields by a warning
+    const hideProperties = []
+    if (import.meta.env.VITE_FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE === 'true' && adapter?.id) {
```

Comment:
question: I've seen this one multiple times maybe worth extracting:

some file

```ts
export const FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE = import.meta.env.VITE_FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE === 'true'
```

`FLAG_ADAPTER_MAPPINGS_IN_WORKSPACE`

Reply:
Nice nudge! I do have a mechanism in place for these... and I forgot about it!
https://github.com/hivemq/hivemq-edge/blob/master/hivemq-edge/src/frontend/src/config/index.ts